src/bin.ts

```diff
@@ -91,6 +103,56 @@ const createSafeS3Key = (key: string): string => {
     return key;
 };
 
+const uploadQueue = queue((task: UploadTask, callback: (key?: string, err?: any) => void) => {
+    const {s3, publicDir, objects, isKeyInUse, path, config, params, spinner} = task;
+    const key = createSafeS3Key(relative(publicDir, path));
+    const stream = fs.createReadStream(path);
+    const hashStream = stream.pipe(createHash('md5').setEncoding('hex'));
+
+    streamToPromise(hashStream)
+        .then((data) => {
```

Comment:
I'd really like to see all usage of .then converted into awaits. It makes the code a lot more readable.